add person sensor (sen21231) from useful sensors

shreyaskarnik opened the pull request:
## Description:


**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1925

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

shreyaskarnik committed:
add person sensor (sen21231) from useful sensors

shreyaskarnik committed:
fix file mode

shreyaskarnik committed:
add image for index

shreyaskarnik committed:
fix image size

jesserockz commented:
As this is a feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase.

Thanks
Jesse

shreyaskarnik committed:
changes from upstream

shreyaskarnik committed:
changes from upstream

shreyaskarnik committed:
Merge branch 'next' into sen21231

shreyaskarnik commented:
Thanks, for the feedback @jesserockz I updated the target branch and rebased but could not fix the change in `_static/version`. Happy to add a new PR for the docs. Apologies about targeting the current branch did not know about the next branch.

jesserockz committed:
Undo change to version file

jesserockz commented on components/sensor/sen21231.rst:
```diff
@@ -0,0 +1,33 @@
+.. code-block:: yaml
+
+    # Example configuration entry
+    output:
```
This is a sensor, not an output...
```suggestion
    sensor:
```

jesserockz commented on components/sensor/sen21231.rst:
```diff
@@ -0,0 +1,33 @@
+      - platform: sen21231
+        id: person_sensor
+        # sda and scl connected approriately
+        # power connected to 3.3v
```
We don't need extra comments here when this information is contained in the i2c docs.
```suggestion
```

jesserockz commented on components/sensor/sen21231.rst:
```diff
@@ -0,0 +1,33 @@
+
+    Person Sensor
+
+This sensor returns information over I2C interface. The I2C address is ``0x62``
```
```suggestion
This sensor requires an :doc:`/components/i2c`. The I2C address is ``0x62``
```

shreyaskarnik committed:
Update components/sensor/sen21231.rst

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

shreyaskarnik committed:
Update components/sensor/sen21231.rst

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

shreyaskarnik committed:
Update components/sensor/sen21231.rst

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

shreyaskarnik commented:
@jesserockz thanks for the suggestions I have applied them. I appreciate your help. 🙌

jesserockz committed:
Update sen21231.rst
